front/lib/api/assistant/agent.ts

```diff
@@ -198,19 +232,25 @@ export async function getNextAction(
     }
   }
 
-  specifications.push({
-    name: "reply_to_user",
-    description:
-      "Reply to the user with a message. You don't need to generate any arguments for this function.",
-    inputs: [],
-  });
+  // TODO(@fontanierh): remove this once generation is part of actions.
+  if (agentConfiguration.generation && isGenerationAllowed) {
+    specifications.push({
+      name: "reply_to_user",
+      description:
+        "Reply to the user with a message. You don't need to generate any arguments for this function.",
+      inputs: [],
+    });
+  }
 
   const config = cloneBaseConfig(
     DustProdActionRegistry["assistant-v2-use-tools"].config
   );
   config.MODEL.function_call = "auto";
   config.MODEL.provider_id = model.providerId;
   config.MODEL.model_id = model.modelId;
+  if (specifications.length === 1) {
```

Comment:
🧐 I'm curious if we should only apply this change for the `forceUseAtIteration` cases? Imagine a scenario where there's only one available action. In this case, we'll always set `function_call` to that action is that expected?